modules/oe_translation_cdt/src/Mapper/TranslationRequestMapper.php

```diff
@@ -0,0 +1,191 @@
+use Drupal\oe_translation_cdt\TranslationRequestCdtInterface;
+use OpenEuropa\CdtClient\Model\Request\Callback;
+use OpenEuropa\CdtClient\Model\Request\File;
+use OpenEuropa\CdtClient\Model\Request\SourceDocument;
+use OpenEuropa\CdtClient\Model\Request\Translation;
+use OpenEuropa\CdtClient\Model\Request\TranslationJob;
+
+/**
+ * Maps the data between the TranslationRequestCDT entity and the DTO.
+ */
+class TranslationRequestMapper implements TranslationRequestMapperInterface {
+
+  protected const CHARACTERS_PER_PAGE = 750;
+
+  protected const VOLUME_MULTIPLIER = 0.5;
+
+  /**
+   * TranslationRequestMapper constructor.
+   *
+   * @param \Drupal\oe_translation_cdt\ContentFormatter\ContentFormatterInterface $contentFormatter
+   *   The content formatter.
+   */
+  public function __construct(
+    protected ContentFormatterInterface $contentFormatter
+  ) {}
+
+  /**
+   * {@inheritdoc}
+   */
+  public function convertEntityToDto(TranslationRequestCdtInterface $entity): Translation {
+    /** @var \Drupal\oe_translation_cdt\TranslationRequestCdtInterface $entity */
+    $translation = new Translation();
+    $translation->setComments((string) $entity->getComments());
+    $translation->setTitle(sprintf(
+      'Translation request #%s',
+      $entity->id()
+    ));
+    $translation->setClientReference((string) $entity->id());
+    $translation->setService('Translation');
+    $translation->setPhoneNumber($entity->getPhoneNumber());
+    $translation->setIsQuotationOnly(FALSE);
+    $translation->setSendOptions('Send');
+    $translation->setPurposeCode('WS');
+    $translation->setDepartmentCode($entity->getDepartment());
+    $translation->setContactUserNames($entity->getContactUsernames());
+    $translation->setDeliveryContactUsernames($entity->getDeliverTo());
+    $translation->setPriorityCode($entity->getPriority());
+    $translation->setCallbacks($this->createCallbacks());
+    $translation->setSourceDocuments([$this->createSourceDocument($entity)]);
+    return $translation;
+  }
+
+  /**
+   * Creates the SourceDocument DTO.
+   *
+   * @param \Drupal\oe_translation_cdt\TranslationRequestCdtInterface $translation_request
+   *   The translation request entity.
+   *
+   * @return \OpenEuropa\CdtClient\Model\Request\SourceDocument
+   *   The source document DTO.
+   */
+  protected function createSourceDocument(TranslationRequestCdtInterface $translation_request): SourceDocument {
+    $source_document = new SourceDocument();
+    $translation_jobs = $this->createTranslationJobs($translation_request);
+    $source_document->setTranslationJobs($translation_jobs);
+    $source_languages = [
+      LanguageCodeMapper::getCdtLanguageCode($translation_request->getSourceLanguageCode(), $translation_request),
+    ];
+    $source_document->setSourceLanguages($source_languages);
+    $source_document->setIsPrivate(FALSE);
+    $source_document->setOutputDocumentFormatCode('XM');
```

Comment:
I guess this should be XML?

Reply:
Actually, it is intended, like [here](https://git.drupalcode.org/project/tmgmt_cdt/-/blob/8.x-1.x/src/Plugin/tmgmt/Translator/TmgmtCdtTranslator.php#L304). The `XML` throws a validation error.